features/step_definitions/deposit_steps.rb

```diff
@@ -62,59 +56,50 @@
 end
 
 Given("a Bentley audio artifact that has not been uploaded") do
-  class ArtifactStub
-    def bag_id
-      "adsladsjadjsklafdjsa"
```

Comment:
I dealt with this by realizing that I was only actually using `@artifact.bag_id` and `@artifact.id`, so I made those two readable to the world instead of the entire `@artifact`. I'm not sure if this is the best way to do it. On one hand, in the 404 case, this makes it a bit clearer that it's a case of the content steward entering an id that we just don't have an object for—but on the other hand, this demonstrates my own lack of knowledge about how chipmunk works.

Is there a case where we do have an artifact, but it hasn't yet been uploaded? If that happens, would a request still 404? Would that be a better thing to test for, and is a typo'd id even worth a scenario? (My vote: `typos cause 404s` doesn't need representation in our feature specifications.)

Reply:
This is another issue with the way package is pseudo-polymorphic. We do create the package before it is uploaded. The POST /requests creates it and puts it in "upload mode".

Where I think we are headed is that, when looking to upload something, we get a Deposit instance that encapsulates its process and status. Then, when a deposit is marked as finished, it starts processing and is attached to a new or existing Artifact.

I agree that "typos cause 404s" is not an acceptance test.

Reply:
To be more directive: If we do `Fabricate(:package)`, it is already "unstored" and "not uploaded". The "not uploaded" portion is really a factor of the upload path on disk being missing/empty. Nothing special about this scenario in that regard.

There was discussion yesterday of how the upload_link is kind of flaky in that the upload and incoming paths have different structures (<username>/incoming vs. incoming/<username>), but that we are current managing that with what amounts to a symlink. It's a side effect invisible to the system that, as soon as something is uploaded, it's in incoming storage -- a bug, to be sure.

Reply:
I think this scenario needs revisiting then, because using `Fabricate(:package)` makes `Check on an artifact that has not yet been uploaded` identical to `Check on an artifact that has been uploaded but isn't preserved yet`. I suggest we combine them into one called `Check on an artifact that has not been preserved yet` (and that we follow the form of the `uploaded but isn't preserved` one but lose the word upload).

Reply:
Ah, good point. I think this was sort of a prospective scenario to start. Go for it!

Also, for what it's worth, if we run into situations where really do want conjunctions, Gherkin also supports `But` to split a step, and the general recommendation is to do so for flexibility in implementing the steps, cf. BDD in Action 5.2.4.